echopype/calibrate/api.py

```diff
@@ -22,6 +19,51 @@
 logger = _init_logger(__name__)
 
 
+def _check_echodata_backscatter_size(echodata: EchoData):
+    """
+    Extracts total gigabytes in `backscatter_r` and `backscatter_i`. If above 2 GiB,
+    then raises warning for how to proceed with the `compute_Sv` workflow without
+    destroying overwhelming the system's memory.
+
+    Parameters
+    ----------
+    echodata : EchoData
+        An `EchoData` object created by using `open_raw` or `open_converted`
+    """
+    # Iterate through Echodata Beam Groups to extract total number of bytes for
+    # backscatter variables
+    total_nbytes = 0
+    ed_group_map = echodata.group_map
+    for key in ed_group_map.keys():
+        echodata_group = ed_group_map[key]["ep_group"]
+        if (
+            echodata_group is not None
+            and echodata_group.startswith("Sonar/Beam_group")
+            and echodata[echodata_group] is not None
+        ):
+            variables = list(echodata[echodata_group].variables.keys())
+            backscatter_variables = [
+                variable for variable in variables if variable.startswith("backscatter_")
+            ]
+            for backscatter_variable in backscatter_variables:
+                total_nbytes += echodata[echodata_group][backscatter_variable].nbytes
+
+    # Compute GigaBytes from Bytes
+    total_gb = total_nbytes / (1024**3)
+
+    # Raise Warning if above 2.0
+    if total_gb > 2.0:
+        logger.warning(
+            "The Echodata Backscatter Variables are large and can cause memory issues. "
+            "Consider modifying compute_Sv workflow: "
+            "Prior to `compute_Sv` run `echodata.chunk(CHUNK_DICTIONARY) and after `compute_Sv` "
+            "run `ds_Sv.to_zarr(ZARR_STORE, compute=True)`. This will ensure that the bulk of the "
+            "computation does not overwhelm your system's memory. The former ensures that "
+            "`compute_Sv` is lazily evaluated and the latter points computation towards a Zarr "
+            "Store in Disk instead of on RAM."
+        )
```

Comment:
```suggestion
    if total_gb > 2.0:
        logger.warning(
            "The Echodata Backscatter Variables are large and can cause memory issues. "
            "Consider modifying compute_Sv workflow: "
            "Prior to `compute_Sv` run `echodata.chunk(CHUNK_DICTIONARY) "
            "and after `compute_Sv` run `ds_Sv.to_zarr(ZARR_STORE, compute=True)`. "
            "This will ensure that the computation is lazily evaluated, "
            "with the results stored directly in a Zarr store on disk, rather then in memory."
        )
```